src/validation.cpp

```diff
@@ -1506,61 +1506,68 @@ bool DisconnectBlock(const CBlock &block, CValidationState &state,
 
     if (pfClean) *pfClean = false;
```

Comment:
If we don't reach ApplyBlockUndo, we still need to set the flag as expected by the caller.